volume/devicemapper/devicemapper.go

```diff
@@ -1160,6 +1197,22 @@ func (v *DeviceMapperVolume) Import(label string, reader io.Reader) error {
 	if err := v.driver.DeviceSet.MountDevice(device, filepath.Join(mountpoint, label), fmt.Sprintf("%s_import", label)); err != nil {
 		return err
 	}
+	// The size of the staging volume is determined by the size of the
+	// tenant volume; we need it to be based off of the dm.basesize value if it
+	// is smaller.
+	curSize, err := v.SizeOf()
```

Comment:
this is TECHNICALLY not valid. The size of the staging volume is determined by the size of the BASE volume; the tenant volume could easily have been resized (and often will be).